src/chemotaxis/g8/Controller.java

```diff
@@ -0,0 +1,120 @@
+package chemotaxis.g8;
+
+import chemotaxis.sim.ChemicalCell;
+import chemotaxis.sim.ChemicalCell.ChemicalType;
+import chemotaxis.sim.ChemicalPlacement;
+import chemotaxis.sim.SimPrinter;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class Controller extends chemotaxis.sim.Controller {
+
+	/**
+	 * Controller constructor
+	 *
+	 * @param start       start cell coordinates
+	 * @param target      target cell coordinates
+	 * @param size     	  grid/map size
+	 * @param grid        game grid/map
+	 * @param simTime     simulation time
+	 * @param budget      chemical budget
+	 * @param seed        random seed
+	 * @param simPrinter  simulation printer
+	 *
+	 */
+	static int[][] DIR = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};
+	ArrayList<Point> shortestPath;
+	ArrayList<Point> selectedCells;
+	int shift;
+	static int INTERVAL = 4;
```

Comment:
I think we should make "INTERVAL" a variable depending on what part of the map we are on (so we can apply more chemicals more frequently on a difficult part of the map) -- but we can do this in the future

Reply:
I agree with you, and we can make our strategy more dynamic in the next version.